Source/Scene/GroundPolylinePrimitive.js

```diff
@@ -483,37 +520,47 @@ define([
+            derivedTilesetCommand.pass = Pass.CESIUM_3D_TILE_CLASSIFICATION;
+            command.derivedCommands.tileset = derivedTilesetCommand;
+
             // derive for 2D
-            var derivedColorCommand = command.derivedCommands.color2D;
-            if (!defined(derivedColorCommand)) {
-                derivedColorCommand = DrawCommand.shallowClone(command);
-                command.derivedCommands.color2D = derivedColorCommand;
-            }
-            derivedColorCommand.vertexArray = vertexArray;
-            derivedColorCommand.renderState = groundPolylinePrimitive._renderState;
-            derivedColorCommand.shaderProgram = groundPolylinePrimitive._sp2D;
-            derivedColorCommand.uniformMap = uniformMap;
-            derivedColorCommand.pass = pass;
-            derivedColorCommand.pickId = 'czm_batchTable_pickColor(v_endPlaneNormalEcAndBatchId.w)';
+            var derived2DCommand = DrawCommand.shallowClone(command, command.derivedCommands.color2D);
+            derived2DCommand.shaderProgram = groundPolylinePrimitive._sp2D;
+            command.derivedCommands.color2D = derived2DCommand;
+
+            var derived2DTilesetCommand = DrawCommand.shallowClone(derivedTilesetCommand, derivedTilesetCommand.derivedCommands.color2D);
+            derived2DTilesetCommand.shaderProgram = groundPolylinePrimitive._sp2D;
+            derivedTilesetCommand.derivedCommands.color2D = derived2DTilesetCommand;
 
             // derive for Morph
-            derivedColorCommand = command.derivedCommands.colorMorph;
-            if (!defined(derivedColorCommand)) {
-                derivedColorCommand = DrawCommand.shallowClone(command);
-                command.derivedCommands.colorMorph = derivedColorCommand;
-            }
-            derivedColorCommand.vertexArray = vertexArray;
-            derivedColorCommand.renderState = groundPolylinePrimitive._renderStateMorph;
-            derivedColorCommand.shaderProgram = groundPolylinePrimitive._spMorph;
-            derivedColorCommand.uniformMap = uniformMap;
-            derivedColorCommand.pass = pass;
-            derivedColorCommand.pickId = 'czm_batchTable_pickColor(v_batchId)';
+            var derivedMorphCommand = DrawCommand.shallowClone(command, command.derivedCommands.colorMorph);
+            derivedMorphCommand.renderState = groundPolylinePrimitive._renderStateMorph;
+            derivedMorphCommand.shaderProgram = groundPolylinePrimitive._spMorph;
+            derivedMorphCommand.pickId = 'czm_batchTable_pickColor(v_batchId)';
+            command.derivedCommands.colorMorph = derivedMorphCommand;
         }
     }
 
+    function updateAndQueueCommand(groundPolylinePrimitive, command, frameState, modelMatrix, cull, boundingVolume, debugShowBoundingVolume) {
+        // Use derived appearance command for morph and 2D
+        if (frameState.mode === SceneMode.MORPHING && command.shaderProgram !== groundPolylinePrimitive._spMorph) {
```

Comment:
Why does this check the shader? Shouldn't the command always be the original or the tileset command?

Reply:
I actually don't know, I copied this from https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Scene/GroundPolylinePrimitive.js#L541-L544. It looks like `GroundPrimitive` doesn't do this sort of check either.

I'll make the change and test that everything still works.

Reply:
2D and morphing seem to work fine still. I pushed the change.